model/trigger/version.go

```diff
@@ -145,3 +123,52 @@ func (t *versionTriggers) generate(sub *event.Subscription) (*notification.Notif
 
 	return notification.New(t.event, sub.Trigger, &sub.Subscriber, payload)
 }
+
+func (t *versionTriggers) versionOutcome(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionSucceeded && t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *versionTriggers) versionFailure(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *versionTriggers) versionSuccess(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionSucceeded {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *versionTriggers) versionRegression(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.VersionFailed {
+		return nil, nil
+	}
+
+	versionTasks, err := task.Find(task.ByVersion(t.version.Id))
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving tasks for version")
+	}
+	count := 0 // TODO: maybe use this count in the message somewhere?
+	for _, t := range versionTasks {
+		shouldNotify, _, err := isTaskRegression(&t)
```

Comment:
Recommend the following:
1. Rename shouldNotify to isRegression.
2. Rename count to shouldNotify, and change to boolean
3. If isRegression is true on even a single task, this notification should be sent so just break at that point and generate.

Reply:
done